lib/editor/story_editor_screen.dart

```diff
@@ -131,91 +106,36 @@ class StoryEditorState extends State<StoryEditorScreen> {
-                      story.actors[id] = Actor(id: id);
-                      translation[id] = ActorTranslation(metaData: meta);
-                    },
-                  ),
-                  tooltip: 'Add actor',
-                  child: const Icon(Icons.person_add_rounded),
-                );
-              case 2:
-                return FloatingActionButton(
-                  onPressed: () => setState(() {
-                    final id = uuid.v4();
-                    story.nodes[id] = Node(id);
-                    translation[id] = MessageTranslation(metaData: meta);
-                  }),
-                  tooltip: 'Add node',
-                  child: const Icon(Icons.add_box_rounded),
-                );
-              default:
-                return const SizedBox();
-            }
-          },
-        ),
-        floatingActionButtonLocation: FloatingActionButtonLocation.endDocked,
-        bottomNavigationBar: BottomAppBar(
-          child: SizedBox(
-            height: kBottomNavigationBarHeight,
-            child: Builder(builder: (context) {
-              final pages = {
-                'General': Icons.auto_stories_rounded,
-                'Actors': Icons.person_rounded,
-                'Nodes': Icons.timeline_rounded,
-              }.entries.toList();
-              return ListView(
-                scrollDirection: Axis.horizontal,
-                children: [
-                  for (var i = 0; i < pages.length; i++) ...[
-                    const SizedBox(width: 8),
-                    IconButton(
-                      tooltip: pages[i].key,
-                      icon: Icon(
-                        pages[i].value,
-                        color: _page == i
-                            ? Theme.of(context).toggleableActiveColor
-                            : null,
-                      ),
-                      onPressed: () => setState(
-                        () {
-                          _page = i;
-                          _pageController.animateToPage(
-                            i,
-                            duration: kTabScrollDuration,
-                            curve: standardEasing,
-                          );
-                        },
-                      ),
-                    ),
-                  ],
-                ],
-              );
-            }),
-          ),
+        bottomNavigationBar: BottomNavigationBar(
+          onTap: (i) => update(() {
+            _page = i;
+            _pageController.animateToPage(
+              i,
+              duration: kTabScrollDuration,
+              curve: standardEasing,
+            );
+          }),
+          currentIndex: _page,
+          items: {
+            'General': Icons.auto_stories_rounded,
+            'Actors': Icons.person_rounded,
+            'Narrative': Icons.timeline_rounded,
+          }
+              .entries
+              .map((p) => BottomNavigationBarItem(
```

Comment:
Why do you sometimes use `.map` and sometimes "collection for"? Let's try to make it consistent unless one of them is really more suitable for the specific situation